hook.py

```diff
@@ -1,17 +1,35 @@
 import logging
+import asyncio
 from pathlib import Path
 from subprocess import Popen
+from ssl import get_server_certificate
 
 name = 'SSL'
 description = 'Run an SSL proxy in front of the server'
-address = None
+address = ''
+
+async def _check_cert():
+    await asyncio.sleep(5)
+    default_cert = ''
+    server_cert = get_server_certificate(('127.0.0.1', 8443))
+    with open(Path('plugins/ssl/conf/insecure_certificate.pem'), 'r') as f:
+        for line in f:
+            if 'PRIVATE KEY' in line:
+                break
+            default_cert += line
```

Comment:
these lines are a different abstraction layer than the function suggests. Since you are reading in the certificate here it'll be cleaner to create a new function read_certificate()